lib/appsignal/finch.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Appsignal.Finch do
+  require Appsignal.Utils
+
+  @tracer Appsignal.Utils.compile_env(:appsignal, :appsignal_tracer, Appsignal.Tracer)
+  @span Appsignal.Utils.compile_env(:appsignal, :appsignal_span, Appsignal.Span)
+  import Appsignal.Utils, only: [module_name: 1]
+
+  @moduledoc false
+
+  require Logger
+
+  def attach do
+    handlers = %{
+      [:finch, :request, :start] => &__MODULE__.finch_request_start/4,
+      [:finch, :request, :stop] => &__MODULE__.finch_request_stop/4
+    }
+
+    for {event, fun} <- handlers do
+      case :telemetry.attach({__MODULE__, event}, event, fun, :ok) do
+        :ok ->
+          _ = Appsignal.Logger.debug("Appsignal.Finch attached to #{inspect(event)}")
+
+          :ok
+
+        {:error, _} = error ->
+          Logger.warn("Appsignal.Finch not attached to #{inspect(event)}: #{inspect(error)}")
+
+          error
+      end
+    end
+  end
+
+  def finch_request_start(
+        _event,
+        _measurements,
+        %{name: name, request: request},
+        _config
+      ) do
+    do_finch_request_start(@tracer.current_span(), name, request)
+  end
+
+  defp do_finch_request_start(nil, _name, _request), do: nil
+
+  defp do_finch_request_start(parent, name, request) do
+    uri = %URI{
+      scheme: Atom.to_string(request.scheme),
+      path: request.path,
+      query: request.query,
+      host: request.host,
+      port: request.port
+    }
+
+    "http_request"
+    |> @tracer.create_span(parent)
+    |> @span.set_name("#{module_name(name)}: #{request.method} #{URI.to_string(uri)}")
+    |> @span.set_attribute("appsignal:category", "request.finch")
```

Comment:
For the request to appear in the "Slow API requests" performance panel, you would have to either [add `"finch"` to this list in the server](https://github.com/appsignal/appsignal-server/blob/19acf365942d4a6460bc40c4244e3613a5c66a34/app/frontend/javascripts/components/EventPage/EventPage.jsx#L64-L72), or change it to `finch.request` instead of `request.finch`.

Reply:
The category `request.finch` is good. I'll update the tracking issue to include this change as well.

The other category `finch.request` doesn't group events in the event breakdown under the "finch" group.